modules/GKB/Release/Steps/MSigDB_GSEA.pm

```diff
@@ -6,24 +6,51 @@ use GKB::Release::Utils;
 use Moose;
 extends qw/GKB::Release::Step/;
 
-has '+gkb' => ( default => "gkbdev" );
+my $gsea_output_file = "Reactome_GeneSet_$version.txt";
+
+has '+gkb' => ( default => 'gkbdev' );
 has '+passwords' => ( default => sub { ['mysql'] } );
 has '+directory' => ( default => "$release/msigdb-gsea" );
-has '+mail' => ( default => sub { 
-					my $self = shift;
-					return {
-						'to' => '',
-						'subject' => $self->name,
-						'body' => "",
-						'attachment' => "Reactome_GeneSet_$version.txt"
-					};
-				}
+has '+mail' => ( default => sub {
+	my $self = shift;
+		return {
+			'to' => '',
+			'subject' => $self->name,
+			'body' => '',
+			'attachment' => $self->directory . '/' . $gsea_output_file
+		};
+	}
 );
 
 override 'run_commands' => sub {
     my ($self, $gkbdir) = @_;
-    
-    $self->cmd("Running GSEA output script", [["./runGSEAOutput.sh $self->{host} $db $user $pass 3306 Reactome_GeneSet_$version.txt 48887 true > msigdb-gsea.out 2> msigdb-gsea.err"]]);
+
+    $self->cmd('Build GSEA jar and dependencies', [['perl build_gsea.pl > build_gsea.out 2> build_gsea.err']]);
+
+    my $port = 3306;
```

Comment:
Why is the port number hardcoded like this?

Reply:
This was a holdover from the arguments passed to the shell script being hard-coded.

I've created a $port variable exported from the Release/Config.pm file.